### PR TITLE
Normalize axis data

### PR DESCRIPTION
**Pull request by miscco**
this is a very hacky try to make datetick compilable for m2t inspired by #905.

The idea is simple. Tacke the rather stupid matlab format and project it into a reasonable range tixz can work with. There is obviously a big drawback, as it is done for all the the axes.

If the xaxis is in datetick one does
```matlab
cleanfigure('transformDates', 'x')
```

### CODE REVIEW

**Comment by egeerardyn** [comment]
This just needs a bit of documentation (in the header of `cleanfigure`) to show how it can be used.

Note also that you can get it to work on a single axes object: `cleanfigure(hAx, ...)` should work without a problem. That is something you could also state in the docs.

**Comment by miscco** [comment]
Yeah, thats true.

However, i thought about epanding it. In general, this has nothing to do with dateticks, but with a normalization of the axis data. So i am inklined to rename it to normalize_axis.

Also i believe i have to change the setup, so that it affects anything that has [upper(axis), 'Data'] in it. You know the figure setup better than me. Would it be sufficient to just go through all the direct children of the axis, or do i have to consider the grandchilds too?

**Comment by egeerardyn** [comment]
@miscco I suppose that you need to handle all ancestors (but I imagine there are quite a lot of edge cases that this cannot handle: e.g. imagine some `hgtransform` somewhere, then you can no longer determine which dimension is affected).

Anyhow, since this is a work-around (and quite experimental), I'm fine with such limitations. Perhaps we should, however, add a warning to the documentation and/or at runtime to warn the users that this is experimental functionality.

**Comment by miscco** [comment]
So i have reworked this to better express, what it is actually doing and added both the comments as well as the warning in the function itself. 

@egeerardyn I have no idea why the build failed. Can you enlighten me?

**Comment by egeerardyn** [comment]
@miscco Jenkins configuration error. My bad (performed an update recently, apparently not the whole config was migrated). Will check out what is happening. Code itself ran fine, BTW.

**Comment by egeerardyn** [comment]
@matlab2tikzBot: please test this

File: src/cleanfigure.m [hunk 1]
```diff
@@ -22,6 +22,13 @@ function cleanfigure(varargin)
 %   Scale the precision the data is represented with. Setting it to 0
 %   or negative values disable this feature.
 %   (default: 1)
+%   CLEANFIGURE('normalizeAxis','xyz',...)
+%   EXPERIMENTAL: Normalize the data of the dimensions spezified by
```
Comment (okomarov) [inline src/cleanfigure.m]: typo: specified

**Comment by egeerardyn** [comment]
@matlab2tikzBot: please test this

**Comment by egeerardyn** [comment]
@miscco I updated Jenkins, this fixed the internal errors which caused the build to fail. Since I think Jenkins is a beast, I will look for alternatives that integrate better with GitHub when I find some time. A possible candidate would be TeamCity.

**Comment by miscco** [comment]
So I found a  bug regarding the TickLabels. Not setting the TickLabelMode to manual changed the TickLabels to their new position. 

I tried a variety of ACID test that seemed to work fine. At least I did not find any corruption in MATLAB.

**Comment by miscco** [comment]
Given this is experimental and we warn about it i will merge this tomorrow if noone objects

File: src/cleanfigure.m [hunk 2]
```diff
@@ -1251,3 +1264,51 @@ function removeNaNs(meta, handle)
...
+                strcmp(val(i), 'y') || ...
+                strcmp(val(i), 'z'));
+    end
+end
+% ========================================================================
+function normalizeAxis(handle, cmdOpts)
+    % Normalizes data from a given axis into the interval [0, 1]
+
+    % Warn about normalizeAxis being experimental
+    warning('cleanfigure:normalizeAxis', ...
```
Comment (egeerardyn) [inline src/cleanfigure.m]: 👍

File: src/cleanfigure.m [hunk 3]
```diff
@@ -1251,3 +1264,51 @@ function removeNaNs(meta, handle)
     bool = isnumeric(val) && ~any(isnan(val)) && (isscalar(val) || numel(val) == 2);
 end
 % =========================================================================
+function bool = isValidAxis(val)
+    bool = length(val) <= 3;
+    for i=1:length(val)
+        bool = bool && ...
+               (strcmp(val(i), 'x') || ...
```
Comment (egeerardyn) [inline src/cleanfigure.m]: `strcmp` -> `strcmpi` (I will take care of this)

**Pull request merged**